Automattic/sensei
# Refactor CSS in Learning Mode

Part of #5540, connected: #5541 

### Changes proposed in this Pull Request

* Move out CSS that's only needed by the default template to `default-theme.scss`
* Move out CSS used by blocks that do not depend on template layout to `sensei-theme-blocks.scss`
  * These are now loaded as the style for all the theme blocks instead
* Move out CSS for the mobile sidebar overlay to only apply when the sidebar toggle block is added
* Tweak all CSS to work with all template variations

### Testing instructions

There are a few cases to test:

- Legacy 4.0.2 variation of the default template: select the new 'Video' template, it's using the 4.0.2 HTML now
- New 4.8.0 template variation: 'Default' template
- New 'Modern' template with the corresponding Sensei Pro branch

Check for each:
- It looks good with various block and non-block themes
- Mobile version works well
- Editable in the site editor
   - (Except the 4.0.2 version that doesn't get its CSS loaded for now, we'll fix that in a different PR)
